### ISSUE TITLE
ome device inventory missing

### ISSUE THREAD

**Issue by ImranJMughal** [dell/dellemc-openmanage-ansible-modules#295]
So I cant remediate config compliance once a new server has been added to OME

My process is: -

Discover (ome_discovery)
add to device group (ome_device_group)
remdiate via (ome_configuration_compliance_baseline)

The problem is that the server configuration collection has happened so the server in OME is at an unknown state.

If I manually do an inventory job (collect configuration inventory and collect driver inventory) it can be remediated

How can I kick off a inventory collection prior to config check and remediation?

![image](https://user-images.githubusercontent.com/22365085/126646597-785c833b-6ebc-46d9-89be-0e633c785dfc.png)

See the newly added device with a question mark against it

**Comment by ImranJMughal** [comment]
Ok so it seems i can only do a config inventory on the device group according to the API? Which module can I use for this?

![image](https://user-images.githubusercontent.com/22365085/126662945-58de236f-58aa-44c4-968d-1fa454e592e9.png)

**Comment by jagadeeshnv** [comment]
@ImranJMughal There is no module for this as of now.

- But once the job is run in UI. You can rerun the job using uri module REST API /api/JobService/Actions/JobService.RunJobs

 https://www.dell.com/support/manuals/en-in/dell-openmanage-enterprise/ome-3.5_omem-1.20.10_apiguide/post-method-for-jobservice.runjobs?guid=guid-0413e3f0-cd5e-4335-b42d-36f0f4d9fda8&lang=en-us 

- To get the JobIds use ome_job_info module

**Comment by ImranJMughal** [comment]
Any idea of when you can implement it? It means we cant update new servers as they have unknown config and driver inv.

**Comment by jagadeeshnv** [comment]
@ImranJMughal Currently we dont have this in the pipeline. Will discuss with team and let you know

**Comment by anupamaloke** [comment]
@ImranJMughal, I have just added an ansible role [ome_device_inventory](https://github.com/dell/redfish-ansible-module/tree/master/roles/ome_device_inventory) to run the inventory task for a Device ID or a Device Service Tag or a OME group name. Could you please try it out and let me know whether it works for you or not?

* Run inventory task for a server device using device ID
```
    - name: "Refresh inventory for a server using ID {{ device_id }}"
      include_role:
        name: ome_device_inventory
      vars:
        ome_hostname: "{{ inventory_hostname }}"
        ome_username: "{{ user }}"
        ome_password: "{{ password }}"
        device_id: 12345
```
* Run inventory task for a server device using service tag
```
    - name: "Refresh inventory for a server using service tag {{ device_service_tag }}"
      include_role:
        name: ome_device_inventory
      vars:
        ome_hostname: "{{ inventory_hostname }}"
        ome_username: "{{ user }}"
        ome_password: "{{ password }}"
        device_service_tag: "123ABCD"
```
* Run inventory task for a static or a query group using group name
```
    - name: "Refresh inventory for group {{ ome_group_name }}"
      include_role:
        name: ome_device_inventory
      vars:
        ome_hostname: "{{ inventory_hostname }}"
        ome_username: "{{ user }}"
        ome_password: "{{ password }}"
        ome_group_name: "OME-Static-Group-A"
```

**Comment by ImranJMughal** [comment]
Hi @anupamaloke - Work well thanks

**Comment by anupamaloke** [comment]
A new module `ome_devices` has been added using which you can:

- Refresh inventory for a device or a list of devices in OME
- Reset iDRAC for a device or a list of devices in OME
- Clear iDRAC job queue for a device or a list of devices in OME

**Comment by anupamaloke** [comment]
Closed by #438
